### PR TITLE
Remove `CXCancellation` (deprecated in 1.1)

### PR DESCRIPTION
**Pull request by 1ucian0**
### Summary

Follow up on #11937, this PR removes the pass `CXCancellation`, which it was deprecated in qiskit 1.1

### CODE REVIEW

File: test/python/transpiler/test_cx_cancellation.py [hunk 1]
```diff
@@ -1,199 +0,0 @@
...
-"""Tests for pass cancelling 2 consecutive CNOTs on the same qubits."""
-
-from qiskit import QuantumRegister, QuantumCircuit
-from qiskit.circuit import Clbit, Qubit
-from qiskit.transpiler import PassManager
-from qiskit.transpiler.passes import CXCancellation
-from test import QiskitTestCase  # pylint: disable=wrong-import-order
-
-
-class TestCXCancellation(QiskitTestCase):
```
Comment (kevinhartman) [inline test/python/transpiler/test_cx_cancellation.py:22]: Would it make sense to move this test class to `test_inverse_cancellation.py` and modify the tests to use `InverseCancellation([CXGate])`?

Comment (1ucian0) [reply test/python/transpiler/test_cx_cancellation.py:22]: good point. Done in fe0e44c824ba3f3e1f133df9e10f0787da50216c . Thanks to that, I found this: https://github.com/Qiskit/qiskit/issues/13437

**Comment by 1ucian0** [comment]
Tagging as blocked by https://github.com/Qiskit/qiskit/issues/13437

**Comment by 1ucian0** [comment]
removing blocker. Thanks @haimeng-zhang!

**Review by ElePT — APPROVED** [review]
LGTM

**Pull request merged**